### ISSUE TITLE
[Blog] - [Search] Missing blog search features

### ISSUE THREAD

**Issue by tbrlpld** [mozilla/foundation.mozilla.org#8978]
As outlined in #8935, the current implementation does not fulfill all the requirements from #8843. 

Specifically, the missing features are the following: 

# Search by language

I do not know how to interpret this requirement and how this could work. It would be helpful for us to discuss what the expected usecases look like (e.g. what search query leads to which results being displayed).

# Search by date

This one is possible but I think get extremely complex very quick, and I did not think this complexity was accounted for when we planned the implementation of the search.

 The problem I see is that there a lot of different ways how dates are written internationally (e.g. ISO: `2022-06-15`, US: `06/15/2022` or `Jun 15, 2022` or `June 15, 2022`, UK: `15 June 2022` or `15th of June 2022`, German: `15.06.2022` or `15. Juni 2022`, ..., I guess you see where  I am going with this). We would need to be able to parse all of them, turn them into a date object and then query the respective blog posts.

Or would we restrict the users to a single format (e.g. the only really standardized format internationally is ISO `2022-06-15`)

Also, would we want people to allow ranges? E.g. only define a month of a year, like `June 2022` (again in all locales)? Or even just a year? Or would we even allow arbitrary ranges with start and end date, like `2022-05-09..2022-06-15`?

Would we require the users to mark exactly what the date is in the search query? E.g. `date:2022-06-15`. Or would we allow them to write a date in the format they want anywhere in the search field. 

# Search ranking

This is was clear in the requirements, but I was surprised to find that the "boost" values on the different search fields did not have the intended effect. For some reason this did not work to sort the search results. Apart from the date and language, all other desired fields are considered for the search, but I was not able to control the ranking. 

The question here would be: how important is the original defined ranking structure to us, and thus how much more effort do we want to invest. This mean creating bug report to Wagtail, maybe try to fix the bug in Wagtail, or try to set up a proper search engine (like Elasticsearch) which should have more features than the currently used simple database search. 

***

@danielfmiranda @nancyt1 Please let me know what your thoughts are on the above points

**Comment by nancyt1** [comment]
@tbrlpld I'm ok to pause on search by language and date for now since I don't have a clear sense of how search will actually be used yet and if that just gets too granular. But for search rankings, I'm wondering if we can test with real content before deciding to invest time into it? I'm not too strict on the original ranking structure as long as the results that come up make sense? 

My concern is just having search results that don't match user expectations of how things should be ordered. Like I know we had feedback on PNI search about how if you search "facebook", the first results aren't facebook products but it seems like they show up because it maybe has facebook somewhere in the body copy. So yeah, just having a little more control over things like that.

**Comment by tbrlpld** [comment]
Totally make sense. I am happy to wait and see how search is going to work on the live site too. We can close this ticket and open one when we feel like there are issue (or reports of issues) with the search once it is implemented. 

Having a word in the title should still make it rank higher than the other fields. But for some reason I can't get body and description ordering right. This might be a bug in Wagtail or a misunderstanding of how I am using it. There definitely is a mechanism to control the ranking. That is the "boost" factor you can put on fields. Just for some, at the moment to me, unclear reason Wagtail does not seem to take these into account as I would have expected. 

Definitely happy to revisit this and try to make this work according to spec with some more time planned for it.

**Comment by tbrlpld** [comment]
I just learned that the Postgres database search only supports 4 boost levels. This means we can only control the ranking at that crude level. If we want finer control, we need to setup a proper search engine. 

The 4 levels explain why I have been seeing unexpected results before. I am now able to control at least the rough ranking. I opted for the following:

1. Title
2. Related fields (topic, author, tags)
3. Search description
4. Body

Does this work for you @nancyt1
